src/Sarif.Multitool/TransformCommand.cs

```diff
@@ -2,21 +2,33 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.IO;
+using System.Text;
 using Microsoft.CodeAnalysis.Sarif.Driver;
 using Microsoft.CodeAnalysis.Sarif.Readers;
 using Microsoft.CodeAnalysis.Sarif.VersionOne;
 using Microsoft.CodeAnalysis.Sarif.Visitors;
+using Microsoft.CodeAnalysis.Sarif.Writers;
 using Newtonsoft.Json;
 
 namespace Microsoft.CodeAnalysis.Sarif.Multitool
 {
-    internal static class TransformCommand
+    internal class TransformCommand
     {
-        public static int Run(TransformOptions transformOptions)
+        private bool _testing;
+        private IFileSystem _fileSystem;
+
+        public TransformCommand(IFileSystem fileSystem = null, bool testing = false)
+        {
+            _fileSystem = fileSystem ?? new FileSystem();
+            _testing = testing;
+        }
+
+        public int Run(TransformOptions transformOptions)
         {
             try
             {
-                if (transformOptions.Version < 1 || transformOptions.Version > 2)
+                if (transformOptions.Version != SarifVersion.OneZeroZero && transformOptions.Version != SarifVersion.Current)
```

Comment:
>Version [](start = 37, length = 7)

Got it, so the Transform command can either downgrade to v1, or upgrade to the latest v2. You can't transform to an intermediate version. Probably worth a comment.

Reply:
oh gee. you make me realize, we should update this code to transform pre-release v2 to current v2 before dropping down to v2. :) this allows ongoing transformation of pre-release v2 to v1. this is what commenting code does for you, makes you think a bit more. :)

---
In reply to: [229510132](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229510132) [](ancestors = 229510132)

Reply:
"pre-release v2 to current v2 to v1", I assume you mean. It's up to you, if you think it's a real scenario. Like, some MS internal tool is generating pre-release v2, but wants to store it in an issue store that still accepts only v1.

---
In reply to: [229516417](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229516417) [](ancestors = 229516417,229510132)

Reply:
right, here's the scenario: i've integrated a pre-release v2-generating tool to my pipeline but need to drop it to v1 for some reason. i

---
In reply to: [229516755](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229516755) [](ancestors = 229516755,229516417,229510132)